packages/blob-index/src/sharded-dag-index.js

```diff
@@ -188,3 +207,63 @@ export const archive = async (model) => {
   const cid = Link.create(dagCBOR.code, digest)
   return ok(CAR.encode({ roots: [{ cid, bytes }], blocks }))
 }
+
+/**
+ * Handles large datasets by processing them in batches to avoid memory issues
+ *
+ * @param {API.ShardedDAGIndex} model
+ * @returns {Promise<API.Result<Uint8Array>>}
+ */
+async function archiveLargeDataset(model) {
+  const blocks = new Map()
+  const index = {
+    content: model.content,
+    shards: /** @type {API.Link[]} */ ([]),
+  }
+
+  // Convert all shards to an array first
+  const allShards = [...model.shards.entries()]
+  const totalShards = allShards.length
+
+  // Process shards in batches
+  for (let i = 0; i < allShards.length; i += ARCHIVE_BATCH_SIZE) {
+    const batch = allShards.slice(i, i + ARCHIVE_BATCH_SIZE)
+    const sortedBatch = batch.sort((a, b) => compare(a[0].digest, b[0].digest))
+
+    for (const s of sortedBatch) {
+      // Process slices in batches
+      const allSlices = [...s[1].entries()]
+      const sortedSlices = []
+
+      // Sort slices in batches
+      for (let j = 0; j < allSlices.length; j += ARCHIVE_BATCH_SIZE) {
+        const sliceBatch = allSlices.slice(j, j + ARCHIVE_BATCH_SIZE)
+        const sortedSliceBatch = sliceBatch.sort((a, b) =>
+          compare(a[0].digest, b[0].digest)
+        )
+        sortedSlices.push(...sortedSliceBatch)
+      }
+
+      // Map the sorted slices
+      const mappedSlices = sortedSlices.map((e) => [e[0].bytes, e[1]])
+
+      const bytes = dagCBOR.encode([s[0].bytes, mappedSlices])
+      const digest = await sha256.digest(bytes)
+      const cid = Link.create(dagCBOR.code, digest)
+      blocks.set(cid.toString(), { cid, bytes })
+      index.shards.push(cid)
+    }
+  }
+
+  // Verify we processed all shards
+  if (index.shards.length !== totalShards) {
```

Comment:
Should be a test not production code.

Reply:
Oops. Sorry. Leftovers from the debugging process.